src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T [] array = (T[]) new  Object [10];
+    private int pointer;
+
     @Override
     public void add(T value) {
-
+        if (pointer == array.length) {
+            grow(array);
+        }
+        array[pointer++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > pointer) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Test failed! Can't correctly add element by index ");
+        }
+        if (index <= pointer) {
+            T[] tmp = (T[]) new Object[array.length - index];
+            for (int i = 0; i < tmp.length; i++) {
+                tmp[i] = array[index + i];
+            }
+            array[index] = value;
+            pointer++;
+            if (pointer == array.length) {
+                grow(array);
+            }
+            for (int i = 0; i < tmp.length - 1; i++) {
+                array[index + i + 1] = tmp[i];
+            }
+        }
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (list.size() < array.length - pointer) {
+            grow(array);
+        }
+        for (int i = 0; i < list.size(); i++) {
+            if (pointer == array.length) {
+                grow(array);
+            }
+            array[pointer++] = list.get(i);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= pointer) {
+            throw new ArrayListIndexOutOfBoundsException("Index does not exist");
+        }
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < 0 || index >= pointer) {
+            throw new ArrayListIndexOutOfBoundsException("Index does not exist");
+        }
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T tmp = null;
```

Comment:
The temporary variable `tmp` is redundant. You can directly return the removed element.